packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -17,30 +17,29 @@ export const preloadMethod = `__vitePreload`
 export const preloadMarker = `__VITE_PRELOAD__`
 
 const preloadHelperId = 'vite/preload-helper'
-const preloadCode = `let scriptRel;const seen = {};export const ${preloadMethod} = ${preload.toString()}`
 const preloadMarkerRE = new RegExp(`"${preloadMarker}"`, 'g')
 
 /**
  * Helper for preloading CSS and direct imports of async chunks in parallel to
  * the async chunk itself.
  */
+
+function detectScriptRel() {
+  // @ts-ignore
+  const relList = document.createElement('link').relList
+  // @ts-ignore
+  return relList && relList.supports && relList.supports('modulepreload')
```

Comment:
Not sure, but TS tells me that `relList.supports` is always present and `relList` is always defined :thinking:

Reply:
So it would be

```ts
  return document.createElement('link').relList.supports('modulepreload')
    ? 'modulepreload'
    : 'preload'
```

Reply:
Same as with the other one, I only split the function but for this PR I prefer not to change the logic inside. We can have other PRs doing this later